File: typescript/src/reverse/reverse.ts
export const iterate: (word: string) => string = (word) => {
  let res = "";
  for (const w of word) {
    res = w + res;
  }
  return res;
};

export const std: (word: string) => string = (word) => {
  return [...word].reverse().join("");
};

File: ruby/src/reverse/reverse.rb
# frozen_string_literal: true

# Reverse :
class Reverse
  def self.iterate(word)
    res = ''
    word.each_char { |w| res = w + res }
    res
  end

  def self.recursive(word)
    return '' if word == ''

    recursive(word[1..]) + word[0]
  end

  def self.std(word)
    word.reverse
  end
end
